src/raw/client.rs

```diff
@@ -1,15 +1,112 @@
 // Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
 
-use super::{BatchDelete, BatchGet, BatchPut, BatchScan, Delete, DeleteRange, Get, Put, Scan};
-use crate::{rpc::RpcClient, BoundRange, Config, Key, KvPair, Result, Value};
+use super::ColumnFamily;
+use crate::{rpc::RpcClient, BoundRange, Config, Error, Key, KvPair, Result, Value};
 
+use derive_new::new;
+use futures::future::Either;
 use futures::prelude::*;
 use futures::task::{Context, Poll};
 use std::{pin::Pin, sync::Arc, u32};
 
+const MAX_RAW_KV_SCAN_LIMIT: u32 = 10240;
+
 /// The TiKV raw [`Client`](Client) is used to issue requests to the TiKV server and PD cluster.
+#[derive(Clone)]
 pub struct Client {
     rpc: Arc<RpcClient>,
+    cf: Option<ColumnFamily>,
+    key_only: bool,
+}
+
+// The macros below make writing `impl Client` concise and hopefully easy to
+// read. Otherwise, the important bits get lost in boilerplate.
+
+// `request` and `scan_request` define public functions which return a future for
```

Comment:
Maybe these can be doc comments on each macro?

Reply:
Do you mean the macro uses or definitions? If the latter, I usually only put doc comments on public API, but I don't mind either way.

Reply:
Hm, okay, let's not worry about it.